File: Add Two Numbers.js
var addTwoNumbers = function(l1, l2) {
    let remainder = 0;
    let head = new ListNode();
    let curr = head;
    
    while (l1 || l2  || remainder) {
        let x = l1 ? l1.val : 0;
        let y = l2 ? l2.val : 0;
        let sum = remainder + x + y;
        remainder = Math.floor(sum/10);
        curr.next = new ListNode(sum % 10);
        
        curr = curr.next;
        if (l1) l1 = l1.next;
        if (l2) l2 = l2.next;
    }
    
    return head.next;
};

File: Add Two Numbers.py
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        sumHead = ListNode()
        curr = sumHead
        carry = 0

        while l1 and l2:
            currSum = l1.val + l2.val + carry
            carry = currSum // 10
            curr.next = ListNode(currSum%10)
            curr = curr.next
            l1 = l1.next
            l2 = l2.next
        
        while l1:
            currSum = l1.val  + carry
            carry = currSum // 10
            curr.next = ListNode(currSum%10)
            curr = curr.next
            l1 = l1.next

        while l2:
            currSum = l2.val  + carry
            carry = currSum // 10
            curr.next = ListNode(currSum%10)
            curr = curr.next
            l2 = l2.next

        if carry:
            curr.next = ListNode(carry)
        
        return sumHead.next
